app/javascript/components/shared/TopHeader.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import TopHeaderDropdownMenu from './TopHeaderDropdownMenu';
+
+class TopHeader extends React.Component {
+  render() {
+    let { username, isAdmin, profileLink, editUserLink, adminLink, logoutLink } = this.props;
+
+    let topHeaderDropdownMenuComponent = () => {
```

Comment:
Why you wrap a component in arrow function? And what purpose to use let?

You can simply pass it to `return()`
```javascript
{
  username && (
   <TopHeaderDropdownMenu
        username={username}
        isAdmin={isAdmin}
        profileLink={profileLink}
        editUserLink={editUserLink}
        adminLink={adminLink}
        logoutLink={logoutLink}
    />
  )
}
```